click_completion/core.py

```diff
@@ -110,7 +110,10 @@ def get_choices(cli, prog_name, args, incomplete):
                     break
     choices = []
     if optctx:
-        choices += [c if isinstance(c, tuple) else (c, None) for c in optctx.type.complete(ctx, incomplete)]
+        choices += [c if isinstance(c, tuple) else(c, None) for c in optctx.type.complete(ctx, incomplete)]
+        if not choices and hasattr(optctx, 'autocompletion') and optctx.autocompletion is not None:
```

Comment:
so it means that the choices provided by the type have the precedence over the autocompletion provided by the user.
Shouldn't it be the reverse? Or maybe merge both?

Reply:
That's the order in the `click` code: https://github.com/pallets/click/blob/93b1699cde5fbafe8a237f8f0d21c8f687b78f2f/click/_bashcomplete.py#L172

Realistically I guess you shouldn't be using both. But certainly for `click.Choice`, I think that should take precedence over the auto-completion, because the input is also validated against the contents of `click.Choice`, so it would be erroneous to be offering any other completions.